File: src/main/java/com/kairlec/service/UserService.java
package com.kairlec.service;

import com.kairlec.pojo.User;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public interface UserService {

    Integer init();

    List<User> getAll();

    User getUser(String username);

    Integer insertUser(User user);

    Integer deleteUser(String username);

    Integer updateLoginInfo(User user);

    Integer updatePassword(User user);
}

File: src/main/java/com/kairlec/service/UserService.kt
package com.kairlec.service

import com.kairlec.pojo.User
import org.springframework.stereotype.Service

@Service
interface UserService {
    fun init(): Int?
    fun all(): List<User>?
    fun getUser(username: String): User?
    fun insertUser(user: User): Int?
    fun deleteUser(username: String): Int?
    fun updateLoginInfo(user: User): Int?
    fun updatePassword(user: User): Int?
}
